src/components/OntologyBrowser/index.tsx

```diff
@@ -123,16 +165,22 @@ class OntologyModal extends React.Component<ModalProps, ModalState> {
     });
   };
 
-  updateData = () => {
-    this.ontologyStore.fetch(this.props.initialSqon).then(() => {
+  updateData = async () => {
+    this.setState({ isLoading: true });
+    const { initialSqon } = this.props;
+    try {
+      await this.ontologyStore.fetch(initialSqon);
       this.transfertDataSource = [];
       this.flattenDataSource(this.ontologyStore.tree as TransferItem[]);
       const newTargetKeys = this.getKeysFromSqon();
       this.setState({
         treeSource: this.ontologyStore.tree,
         targetKeys: newTargetKeys,
+        isLoading: false,
       });
-    });
+    } catch (e) {
```

Comment:
Maybe it's better to just wrap the try/catch in what might cause the exception. It's better pratice to scope down so it clear in the future what might throw

Reply:
Very good point